data-serving/scripts/export-data/transform.py

```diff
@@ -220,13 +224,88 @@ def convert_row(row: dict[str, Any]) -> Optional[dict[str, Any]]:
     return row
 
 
+class Writer:
+    "Generic writer class for different formats"
+
+    VALID_FORMATS = ["csv", "tsv", "json", "jsonl"]
+
+    def __init__(self, fmt, output, fields):
+        self.fmt = fmt
+        if fmt not in self.VALID_FORMATS:
+            raise ValueError(f"Invalid format passed to Writer: {fmt}")
+        self.file = gzip.open(f"{output}.{fmt}.gz", "wt")
+        if fmt == "csv":
+            self.writer = csv.DictWriter(
+                self.file, fieldnames=fields, extrasaction="ignore"
+            )
+        elif fmt == "tsv":
+            self.writer = csv.DictWriter(
+                self.file, fieldnames=fields, extrasaction="ignore", delimiter="\t"
+            )
+        else:
+            self.writer = None
```

Comment:
Better to use a context manager (i.e. `with open()...`) and error handling. I also think this code becomes cleaner _without_ the use of classes.

Reply:
Does seem this is a candidate for Strategy pattern, as this `if fmt==…` gets replicated in multiple methods.

Reply:
Thanks for the contextmanager suggestion, and reworking it without classes has made the code clearer!